### PR TITLE
[ui] Service Discovery: Allocation Service fly-out

### LINKED ISSUE #14386: [ui] Service Discovery: Service fly-out detail view
image

### PR DESCRIPTION
**Pull request by philrenaud**
resolves #14386

Establishes a `sidebar` component with queryParam persistence.

### Side-effect
- Adds the concept of a `refID` to service fragments for easier lookup. They already have a `TaskName` property, and added a `GroupName` property as well.

### CODE REVIEW

File: ui/app/styles/components/sidebar.scss [hunk 1]
```diff
@@ -1,3 +1,6 @@
+$topNavOffset: 112px;
+$subNavOffset: 49px;
```
Comment (philrenaud) [inline ui/app/styles/components/sidebar.scss:2]: Making these variables to avoid magic-number syndrome

File: ui/app/serializers/task-group.js [hunk 2]
```diff
@@ -8,6 +8,11 @@ export default class TaskGroup extends ApplicationSerializer {
   mapToArray = ['Volumes'];
 
   normalize(typeHash, hash) {
+    if (hash.Services) {
+      hash.Services.forEach((service) => {
+        service.GroupName = hash.Name;
+      });
+    }
```
Comment (philrenaud) [inline ui/app/serializers/task-group.js:15]: not needed in the task serializer, as services there already get populated with `TaskName` props.

**Review by ChaiWithJai — APPROVED** [review]

File: ui/app/components/allocation-service-sidebar.hbs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+				<div>
+					<span class="pair">
+						<span class="term">
+							Allocation
+						</span>
+						<LinkTo
+							@route="allocations.allocation"
+							@model={{@allocation}}
+							@query={{hash service=""}}
```
Comment (ChaiWithJai) [inline ui/app/components/allocation-service-sidebar.hbs:49]: `question`: Is this for resetting the query parameter?

File: ui/app/components/evaluation-sidebar/detail.js [hunk 4]
```diff
@@ -75,12 +75,4 @@ export default class Detail extends Component {
...
     return this.statechart.send('MODAL_CLOSE');
   }
-
-  keyCommands = [
-    {
-      label: 'Close Evaluations Sidebar',
-      pattern: ['Escape'],
-      action: () => this.closeSidebar(),
-    },
-  ];
```
Comment (ChaiWithJai) [inline ui/app/components/evaluation-sidebar/detail.js:85]: `question` What's this doing here? Do we want to remove this behavior?

File: ui/app/components/service-status-bar.js [hunk 5]
```diff
@@ -8,22 +8,19 @@ import classic from 'ember-classic-decorator';
...
+    if (!this.status) {
       return [];
     }
 
-    const pending = service.pending || 0;
-    const failing = service.failure || 0;
-    const success = service.success || 0;
+    const pending = this.status.pending || 0;
+    const failing = this.status.failure || 0;
+    const success = this.status.success || 0;
```
Comment (ChaiWithJai) [inline ui/app/components/service-status-bar.js:23]: `praise` Good job here. Just be mindful to update the mocks in the test spec now.

File: ui/app/controllers/allocations/allocation/index.js [hunk 6]
```diff
@@ -67,38 +71,32 @@ export default class IndexController extends Controller.extend(Sortable) {
...
+        healthChecks.forEach((check) => {
+          if (
+            !service.healthChecks.find(
+              (sc) =>
+                sc.Check === check.Check && sc.Timestamp === check.Timestamp
+            )
+          ) {
+            service.healthChecks.pushObject(check);
+          }
+        });
```
Comment (ChaiWithJai) [inline ui/app/controllers/allocations/allocation/index.js:96]: `reminder`: Update tests for this API change in the unit tests for this function.

File: ui/app/models/service-fragment.js [hunk 7]
```diff
@@ -11,8 +14,34 @@ export default class Service extends Fragment {
...
+  @attr({ defaultValue: () => [] }) healthChecks;
+
+  @computed('healthChecks.[]')
+  get mostRecentChecks() {
+    // Get unique check names, then get the most recent one
+    return this.get('healthChecks')
+      .mapBy('Check')
+      .uniq()
+      .map((name) => {
+        // Assumtion: health checks are being pushed in sequential order (hence .reverse)
```
Comment (ChaiWithJai) [inline ui/app/models/service-fragment.js:29]: `nit`: typo `Assumtion` is missing a `p`.

File: ui/app/templates/allocations.hbs [hunk 8]
```diff
@@ -1,3 +1,4 @@
 <PageLayout>
+  {{!-- <PortalTarget @name="service-detail-portal" /> --}}
```
Comment (ChaiWithJai) [inline ui/app/templates/allocations.hbs:2]: `nit` Do we want to commit this code? Its commented rn.

**Review by ChaiWithJai — APPROVED** [review]
LGTM

File: ui/app/components/allocation-service-sidebar.hbs (hunk 3 above)
Comment (philrenaud) [reply ui/app/components/allocation-service-sidebar.hbs:49]: Yep!

File: ui/app/components/evaluation-sidebar/detail.js (hunk 4 above)
Comment (philrenaud) [reply ui/app/components/evaluation-sidebar/detail.js:85]: Ack, great catch! I did not mean to remove this; probably cut when I meant to copy.

**Pull request merged**